src/towncrier/test/test_build.py

```diff
@@ -98,17 +86,72 @@ def test_command(self):
     def test_subcommand(self):
         self._test_command(_main)
 
-    def test_no_newsfragment_directory(self):
+    @with_isolated_runner
+    def test_in_different_dir_dir_option(self, runner):
         """
-        A missing newsfragment directory acts as if there are no changes.
+        The current working directory doesn't matter as long as we pass
+        the correct one.
         """
-        runner = CliRunner()
+        project_dir = Path(".").resolve()
 
-        with runner.isolated_filesystem():
-            setup_simple_project()
-            os.rmdir("foo/newsfragments")
+        setup_simple_project()
+        Path("foo/newsfragments/123.feature").write_text("Adds levitation")
+        # Ensure our assetion below is meaningful.
+        self.assertFalse((project_dir / "NEWS.rst").exists())
 
-            result = runner.invoke(_main, ["--draft", "--date", "01-01-2001"])
+        # Create a temporary directory, run Towncrier from there and assert
+        # it didn't litter into it.
+        with tempfile.TemporaryDirectory() as td:
+            os.chdir(td)
+            result = runner.invoke(cli, ("--yes", "--dir", str(project_dir)))
+
+            self.assertEqual([], list(Path(td).glob("*")))
+            # Staying in the temporary directory break cleanup on Windows.
+            os.chdir(project_dir)
```

Comment:
This probably needs to be in a finally or done with `addCleanup` or a failure from that `assertEqual` just above will cause problems (otoh why doesn't trial do this for you)

Reply:
hmmm what you're saying is to use the non-contextmanager version…didn't think of that, I'll change it. Saves an indentation level too!

Reply:
ah much better – thanks!